### DISCUSSION TITLE
Inconsistent: UserManager.Get.... and UserManager.Get....Async take different arguments

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/aspnetcore#27168] [General] [answered]
User manager get methods accept a different object depending on whether it is async or sync.

E.g. GetUserId wants a ClaimsPrincipal
GetUserIdAsync wants an application user class.

Is there a good reason for this or should these be brought into harmony to use ClaimsPrincipal perhaps?

**Comment by davidfowl** [comment]
@HaoK @blowdart

**Reply by blowdart** [reply]
History probably. Couldn't change it now, would be way too breaking.

**Comment by HaoK** [comment, marked as answer]
They also are used for very different things as well, one is used to extract the user id from an existing principal, i.e. HttpContext.User, while GetUserId is returning the user id from the user poco.
